### PR TITLE
Fixes erroneous Bad Dreams ability popup.

### PR DESCRIPTION
**Pull request by ShaeTsuPog**
## Description
Currently on Upcoming the ability popup for Bad Dreams triggers even when no Pokemon are asleep. This PR fixes that by adding a loop over all battlers that checks for sleep before triggering the battlescript.

## Images
Before:
baddreamsbug

After, featuring a double battle with scuffed gameplay:
baddreamsfix

## Issue(s) that this PR fixes
N/A

## **Discord contact info**
Shae#9032

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -4974,8 +4974,14 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
...
-                BattleScriptPushCursorAndCallback(BattleScript_BadDreamsActivates);
-                effect++;
+                for (i = 0; i < MAX_BATTLERS_COUNT; i++)
+                {
+                    if (gBattleMons[i].status1 & STATUS1_SLEEP)
+                    {
+                        BattleScriptPushCursorAndCallback(BattleScript_BadDreamsActivates);
+                        effect++;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (ShaeTsuPog) [reply src/battle_util.c]: Doing it this way seems to create a noticeable lag spike when the ability popup slides away.

laggy

[the remark this answers is not in the document]
Comment (ShaeTsuPog) [reply src/battle_util.c]: Also causes this if a mon dies from bad dreams.

mGBA_9ODsTJiakn

**Comment by AlexOn1ine** [comment]
I am not sure but I think Bad Dreams shouldn't effect partner pokemon if I understand bulbagarden correctly.

> At the end of each turn, each adjacent sleeping opponent takes damage equal to 1/8 of its maximum HP.

**Comment by LOuroboros** [comment]
> I am not sure but I think Bad Dreams shouldn't effect partner pokemon if I understand bulbagarden correctly.
> 
> > At the end of each turn, each adjacent sleeping opponent takes damage equal to 1/8 of its maximum HP.

Yeah, and it doesn't. Or at least it didn't after I was done with the rework and it was merged. The battle script should filter out the user's partner.

https://github.com/rh-hideout/pokeemerald-expansion/blob/upcoming/data/battle_scripts_1.s#L8926

**Comment by mrgriffin** [comment]
Superseded by #3131.

**Pull request closed without merging**